app/views/unattended/provisioning_templates/snippet/puppet.conf.erb

```diff
@@ -47,10 +47,14 @@ pluginsync      = true
 report          = true
 <%- if host_puppet_ca_server.present? -%>
 ca_server       = <%= host_puppet_ca_server %>
+<%- elsif host_param('puppet_ca_server') != '' -%>
```

Comment:
As you can see from the failed tests this results in `true` because `nil != ''`. For that we have the `.present?` method:
```suggestion
<%- elsif host_param('puppet_ca_server').present? -%>
```